hilti/toolchain/src/compiler/parser/parser.yy

```diff
@@ -1,7 +1,7 @@
 /* Copyright (c) 2020-2021 by the Zeek Project. See LICENSE for details. */
 
 %skeleton "lalr1.cc"                          /*  -*- C++ -*- */
-%require "3.4"
+%require "3.3"
```

Comment:
I'm wondering if we should set this to 3.0 as that's "mostly true", and avoids confusion by somebody looking just at this file. And we can add comment to the `api.parser.class` class line that it'll be replaced for < 3.3.

Alternatively, I'd at least a comment here saying that this 3.3 actually is a 3.0 through PP.

Reply:
I wanted to avoid the grammar being inconsistent (e.g., incompatibility between version requirement and used API), so the choice here was to either declare a 3.0 requirement and preprocess to use a 3.3 API if detected, or the conditonal downgrade done here. I went with the downgrade since I expect platforms to upgrade their Bison packages so that we can remove the downgrade support without code changes.

I'll go with a comment explaining that we might downgrade the file (here and in the other grammar),

```
/* This grammar is written against the bison-3.3 API. If an older Bison version
 * was detected we perform preprocessing to support versions down to at least
 * bison-3.0, see the CMake macro `BISON_TARGET_PP`. */
%require "3.3"
```